app/elements/submission/tasks/canvas-game-task.html

```diff
@@ -0,0 +1,350 @@
+			width: 100%;
+		}
+		#leftCard{
+			margin: 16px 8px 16px 16px;
+			width: calc(100% - 24px);
+			height: calc(100% - 32px);
+			overflow: auto;
+		}
+		#topRightCard>h2{
+			height: 40px;
+			margin: 0;
+			padding-left: 16px;
+		}
+		#topRightCard{
+			overflow: auto;
+			margin: 16px 16px 16px 8px;
+			width: calc(100% - 24px);
+			height: calc(50% - 24px);
+		}
+		#botRightCard{
+			overflow: hidden;
+			margin: 8px 16px 16px 8px;
+			width: calc(100% - 24px);
+			height: calc(50% - 24px);
+		}
+		#buttons{
+			height: 45px;
+			margin: 16px 16px 16px 8px;
+		}
+		#consoleWrapper{
+			margin: 8px 16px 16px 16px;
+			height: calc(100% - 90px);
+		}
+		#buttons>paper-icon-button{
+			padding: 5px;
+			margin: 0;
+		}
+	</style>
+	<template>
+		<iron-ajax
+			id="ajaxRequest"
+			handle-as="json"
+			debounce-duration="1000"
+			content-type="application/json"
+			>
+		</iron-ajax>
+		<iron-ajax
+			id="getTemplateAjax"
+			handle-as="text"
+			on-response="onTemplateResponse"
+			on-error="onTemplateError"
+			>
+		</iron-ajax>
+		<div id="componentWrapper" class="horizontal layout">
+			<div class="flex">
+				<paper-material id="leftCard" elevation="1">
+					<canvas id="canvas" width="{{canvasWidth}}" height="{{canvasHeight}}"></canvas>
+				</paper-material>
+			</div>
+			<div class="flex">
+				<paper-material id="topRightCard" elevation="1">
+					<div id="codeEditorWrapper">
+						<code-editor id="codeEditor"></code-editor>
+					</div>
+				</paper-material>
+				<paper-material id="botRightCard" elevation="1">
+					<div id="buttons" class="horizontal layout">
+						<paper-icon-button icon="av:play-arrow" on-click="runCode">Run code</paper-icon-button>
+						<span class="flex"></span>
+						<paper-button raised noink class$="{{finishedButtonClass}}" on-click="finishTask">Finish</paper-button>
+					</div>
+					<div id="consoleWrapper">
+						 <task-display task="{{task}}"></task-display>
+					</div>
+				</paper-material>
+			</div>
+		</div>
+	</template>
+	<script>
+		Polymer({
+			// TODO: Extract canvas into own component
+			is: 'canvas-game-task',
+			properties: {
+				obstacles: {
+					type: Array
+				},
+				coins: {
+					type: Array
+				},
+				moves: {
+					type: Array
+				},
+				maxPosition: {
+					type: Object
+				},
+				finishPosition: {
+					type: Object
+				},
+				oldPosition: {
+					type: Object
+				},
+				startPosition: {
+					type: Object
+				},
+				xRatio: {
+					type: Number
+				},
+				yRatio: {
+					type: Number
+				},
+				counter: {
+					type: Number,
+					value: 0
+				},
+				task: {
+					type: Object,
+					notify: true,
+					reflectToAttribute: true
+				}
+			},
+			observers: [
+				'_afterPropertiesSet(task)'
+			],
+			listeners: {
+				'ws-onerror': 'onWsError',
+				'ws-onmessage': 'onWsMessage',
+				'ws-onopen': 'onWsOpen'
+			},
+			onWsOpen: function(){
+				console.log('WS OPENED');
+			},
+			onWsError: function(err){
+				console.log(err);
```

Comment:
we don`t currently have an util for handling errors from the websockets. I`ll code it sometime this week.